src/monitor/monitor.c

```diff
@@ -1973,46 +1918,86 @@ static void monitor_restart_service(struct mt_svc *svc)
     }
 }
 
+static void check_nscd(void)
+{
+    int ret;
+    ret = check_file(NSCD_SOCKET_PATH,
+                     -1, -1, S_IFSOCK, S_IFMT, NULL, false);
+    if (ret == EOK) {
+        ret = sss_nscd_parse_conf(NSCD_CONF_PATH);
+
+        switch (ret) {
+            case ENOENT:
+                sss_log(SSS_LOG_NOTICE,
+                        "NSCD socket was detected. NSCD caching capabilities "
+                        "may conflict with SSSD for users and groups. It is "
+                        "recommended not to run NSCD in parallel with SSSD, "
+                        "unless NSCD is configured not to cache the passwd, "
+                        "group, netgroup and services nsswitch maps.");
+                break;
+
+            case EEXIST:
+                sss_log(SSS_LOG_NOTICE,
+                        "NSCD socket was detected and seems to be configured "
+                        "to cache some of the databases controlled by "
+                        "SSSD [passwd,group,netgroup,services]. It is "
+                        "recommended not to run NSCD in parallel with SSSD, "
+                        "unless NSCD is configured not to cache these.");
+                break;
+
+            case EOK:
+                DEBUG(SSSDBG_TRACE_FUNC, "NSCD socket was detected and it "
+                            "seems to be configured not to interfere with "
+                            "SSSD's caching capabilities\n");
+        }
+    }
+}
+
+int bootstrap_monitor_process(uid_t target_uid, gid_t target_gid);
+void setup_keyring(void);
+
 int main(int argc, const char *argv[])
 {
     int opt;
     poptContext pc;
     int opt_daemon = 0;
     int opt_interactive = 0;
-    int opt_genconf = 0;
     int opt_version = 0;
-    int opt_netlinkoff = 0;
     char *opt_config_file = NULL;
     const char *opt_logger = NULL;
     char *config_file = NULL;
-    char *opt_genconf_section = NULL;
-    int flags = 0;
+    int flags = FLAGS_NO_WATCHDOG;
     struct main_context *main_ctx;
     TALLOC_CTX *tmp_ctx;
     struct mt_ctx *monitor;
     int ret;
-    uid_t uid;
+    uid_t uid, euid;
+    gid_t gid, egid;
+    char *initial_caps;
+    struct sss_ini *config;
+    char *cdb_file = NULL;
+
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        return 2;
+    }
+    monitor = talloc_zero(tmp_ctx, struct mt_ctx);
+    if (monitor == NULL) {
+        return 2;
```

Comment:
Hi,

we currently do not call `talloc_free(tmp_ctx)` when exiting after an error. Do you think it is worth to add it for the sake of a clean process termination?

bye,
Sumit

Reply:
Done.

Reply:
Thanks.